subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -405,9 +735,250 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
+
+	bt_request_cb = cb;
+}
+
+int fp_storage_ak_bond_conn_create(const void *conn_ctx, const bt_addr_le_t *addr,
+				   const struct fp_account_key *account_key)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return -ENOTSUP;
+	}
+
+	if (!is_enabled) {
+		return -EACCES;
+	}
+
+	int err;
+	uint8_t id;
+	struct fp_bond *bond;
+	struct fp_bond bond_rollback_copy = (struct fp_bond){0};
+
+	if (account_key) {
+		err = ak_id_get(&id, account_key);
+		if (err) {
+			return err;
+		}
+	} else {
+		/* The actual ID will be determined during Account Key write. */
+		id = BOND_AK_ID_UNKNOWN;
+	}
+
+	bond = fp_bond_free_get();
+	if (!bond) {
+		return -ENOMEM;
+	}
+
+	bond->conn_ctx = conn_ctx;
+	bond->bonded = false;
+	bond->nv.account_key_metadata_id = id;
+	bond->nv.addr = *addr;
+
+	err = fp_bond_settings_save(bond, &bond_rollback_copy);
+	if (err) {
+		return err;
+	}
+
 	return 0;
 }
 
+void fp_storage_ak_bond_conn_confirm(const void *conn_ctx, const bt_addr_le_t *addr)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+
+	if (!is_enabled) {
+		return;
+	}
+
+	struct fp_bond *bond;
+	struct fp_bond *bond_duplicate;
+	int err;
+
+	bond = fp_bond_get(conn_ctx);
+	if (!bond) {
+		return;
+	}
+
+	bond->bonded = true;
+
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		bond_duplicate = bonded_duplicate_get(conn_ctx, addr);
+		if (bond_duplicate) {
+			err = fp_bond_delete(bond_duplicate);
+			if (err) {
+				LOG_ERR("Failed to delete bond (err %d)", err);
+				break;
+			}
+
+			LOG_DBG("Deleted duplicated bond");
+		} else {
+			break;
+		}
+	}
+
+	if (bond->nv.account_key_metadata_id) {
+		/* Account Key already associated (subsequent pairing procedure).
+		 * Procedure finished successfully. Setting conn_ctx to NULL.
+		 */
+		bond->conn_ctx = NULL;
+	}
+}
+
+void fp_storage_ak_bond_conn_addr_update(const void *conn_ctx, const bt_addr_le_t *new_addr)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+
+	if (!is_enabled) {
+		return;
+	}
+
+	bond_update_addr(conn_ctx, new_addr);
+}
+
+void fp_storage_ak_bond_conn_cancel(const void *conn_ctx)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+
+	if (!is_enabled) {
+		return;
+	}
+
+	struct fp_bond *bond;
+	int err;
+
+	bond = fp_bond_get(conn_ctx);
+	if (!bond) {
+		return;
+	}
+
+	/* If at this point the conn_ctx is not set to NULL, the procedure is incomplete. */
+	err = bond_remove_completely(bond);
+	if (err) {
+		LOG_ERR("Failed to remove bond (err %d)", err);
+	}
+}
+
+void fp_storage_ak_bond_delete(const bt_addr_le_t *addr)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+
+	if (!is_enabled) {
+		return;
+	}
+
+	struct fp_bond *bond;
+	int err;
+
+	LOG_DBG("Bluetooth bond deleted");
+	bond = bonded_by_addr_get(addr);
```

Comment:
`addr_to_fp_bond_info` ?

Reply:
`fp_bond_info_by_addr_bonded_get` we only get bonded ones